include/SDL3/SDL_filesystem.h

```diff
@@ -264,11 +264,42 @@ typedef Uint32 SDL_GlobFlags;
  */
 extern SDL_DECLSPEC bool SDLCALL SDL_CreateDirectory(const char *path);
 
-/* Callback for directory enumeration. Return 1 to keep enumerating,
-   0 to stop enumerating (no error), -1 to stop enumerating and
-   report an error. `dirname` is the directory being enumerated,
-   `fname` is the enumerated entry. */
-typedef int (SDLCALL *SDL_EnumerateDirectoryCallback)(void *userdata, const char *dirname, const char *fname);
+/**
+ * Possible results from an enumeration callback.
+ *
+ * \since This enum is available since SDL 3.0.0.
+ *
+ * \sa SDL_EnumerateDirectoryCallback
+ */
+typedef enum SDL_EnumResult
+{
+    SDL_ENUM_CONTINUE,   /**< Value that requests that enumeration continue. */
+    SDL_ENUM_SUCCESS,    /**< Value that requests that enumeration stop, successfully. */
+    SDL_ENUM_FAILURE     /**< Value that requests that enumeration stop, as a failure. */
+} SDL_EnumResult;
```

Comment:
```suggestion
} SDL_EnumerationResult;
```

"Is this a typedef enum Result, or an enumeration result?"

Reply:
That's fine, but let's keep the symbols as `SDL_ENUM_*` to keep it short.

Reply:
Changed in e1a2edb18898d7b90b4da2a64d590f1b9d35c0dc.

Reply:
> That's fine, but let's keep the symbols as `SDL_ENUM_*` to keep it short.

Agreed.